### PR TITLE
add --dist and --scl to `obal srpm`

### PR DESCRIPTION
**Pull request by komidore64**

### CODE REVIEW

File: obal/data/roles/build_srpm/tasks/main.yaml [hunk 1]
```diff
@@ -1,10 +1,41 @@
 ---
+- name: 'Set build_srpm_tito_releaser_args'
+  set_fact:
+    build_srpm_tito_releaser_args: "{{ build_package_tito_releaser_args | default([]) }}"
```
Comment (evgeni) [inline obal/data/roles/build_srpm/tasks/main.yaml]: you define `build_srpm_tito_releaser_args: []` in `defaults`, which makes me think the user is allowed to set `build_srpm_tito_releaser_args: ['--whatever']`. but this `set_fact` will override it?

File: obal/data/roles/build_srpm/defaults/main.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+build_srpm_tito_releaser_args: []
+build_srpm_dist_args: ''
```
Comment (evgeni) [inline obal/data/roles/build_srpm/defaults/main.yaml]: what is this (and the next one) for? I don't see the vars used.

File: obal/data/roles/build_srpm/tasks/main.yaml (hunk 1 above)
Comment (komidore64) [reply obal/data/roles/build_srpm/tasks/main.yaml]: Hrm, yeah that does seem a bit goofy. Let me clean this up.

File: obal/data/roles/build_srpm/defaults/main.yaml (hunk 2 above)
Comment (komidore64) [reply obal/data/roles/build_srpm/defaults/main.yaml]: removed

File: obal/data/roles/build_srpm/tasks/main.yaml (hunk 1 above)
Comment (komidore64) [reply obal/data/roles/build_srpm/tasks/main.yaml]: defaults to `build_package_tito_releaser_args` but overrides with passed in `--dist` and/or `--scl` flags, or `build_srpm_tito_releaser_args` if you like extra vars.

**Comment by komidore64** [comment]
rebased

File: obal/data/roles/build_srpm/tasks/main.yaml [hunk 3]
```diff
@@ -1,10 +1,36 @@
 ---
+- block:
+    - name: 'Set dist from commandline args'
+      set_fact:
+        build_srpm_tito_releaser_args: "{{ build_srpm_tito_releaser_args | default([]) }} + [ '--dist {{ build_srpm_dist }}' ]"
```
Comment (evgeni) [inline obal/data/roles/build_srpm/tasks/main.yaml]: given `build_srpm_tito_releaser_args` is defined in `defaults`, I think we don't need the `| default([])` here?

File: obal/data/roles/build_srpm/tasks/main.yaml [hunk 4]
```diff
@@ -1,10 +1,36 @@
...
+      when: build_srpm_dist is defined
+
+    - name: 'Set scl from commandline args'
+      set_fact:
+        build_srpm_tito_releaser_args: "{{ build_srpm_tito_releaser_args | default([]) }} + [ '--scl {{ build_srpm_scl }}' ]"
```
Comment (evgeni) [inline obal/data/roles/build_srpm/tasks/main.yaml]: given `build_srpm_tito_releaser_args` is defined in `defaults`, I think we don't need the `| default([])` here?

File: tests/fixtures/mockbin/mockbin [hunk 5]
```diff
@@ -45,10 +48,12 @@ if prog == 'tito':
     parser_release.add_argument('-y', action='store_const', dest='output', const='Created task: 1234')
     parser_release.add_argument('--arg')
     parser_build = subparsers.add_parser('build')
-    parser_build.add_argument('--srpm', action='store_const', dest='output', const='Wrote: hello.src.rpm')
+    parser_build.add_argument('--srpm', action='store_const', dest='output', const="Wrote: {}".format(_FAKE_SRPM_PATH))
     parser_build.add_argument('--scl', action='store')
     parser_build.add_argument('--offline', action='store_true')
     args = parser.parse_args()
+    if 'output' in args and re.compile(r'Wrote').search(args.output):
+        open(_FAKE_SRPM_PATH, 'w').close()
```
Comment (evgeni) [inline tests/fixtures/mockbin/mockbin]: `/tmp/tito` does not exist at this point, so you have to `os.mkdir` it first (and catch the case where it actually does exist, like on all our workstations).

File: tests/fixtures/mockbin/mockbin [hunk 6]
```diff
@@ -45,10 +48,12 @@ if prog == 'tito':
     parser_release.add_argument('-y', action='store_const', dest='output', const='Created task: 1234')
     parser_release.add_argument('--arg')
     parser_build = subparsers.add_parser('build')
-    parser_build.add_argument('--srpm', action='store_const', dest='output', const='Wrote: hello.src.rpm')
+    parser_build.add_argument('--srpm', action='store_const', dest='output', const="Wrote: {}".format(_FAKE_SRPM_PATH))
     parser_build.add_argument('--scl', action='store')
     parser_build.add_argument('--offline', action='store_true')
     args = parser.parse_args()
+    if 'output' in args and re.compile(r'Wrote').search(args.output):
```
Comment (evgeni) [inline tests/fixtures/mockbin/mockbin]: I think you can just write `if 'output' in args and _FAKE_SRPM_PATH in args.output`

**Comment by komidore64** [comment]
@evgeni rebased and updated

File: obal/data/playbooks/srpm/metadata.obal.yaml [hunk 7]
```diff
@@ -3,3 +3,13 @@ help: >
 
 
   Building source RPMs can be useful when locally building packages.
+variables:
+  build_srpm_dist:
+    parameter: --dist
+    help: Dist to set when building a SRPM
+  build_srpm_scl:
```
Comment (ekohl) [inline obal/data/playbooks/srpm/metadata.obal.yaml]: I thought SRPMs weren't aware of SCL since it's just a combination of source + spec.

Comment (komidore64) [reply obal/data/playbooks/srpm/metadata.obal.yaml]: They technically aren't SCL aware, but tito can still prepend the scl prefix to the front of the SRPM name.

**Comment by komidore64** [comment]
@ekohl removed

**Comment by ekohl** [comment]
@komidore64 I meant to create a PR with only the ansible-lint fixes so we could merge that. This one needs squashing anyway :)

**Comment by evgeni** [comment]
bleh, the travis errors are unrelated, your change looks good

**Comment by komidore64** [comment]
squashed

**Pull request merged**